aten/src/ATen/native/Linear.cpp

```diff
@@ -41,6 +41,23 @@ Tensor linear(const Tensor& input, const Tensor& weight, const c10::optional<Ten
   return output;
 }
```

Comment:
Can we refactor `linear` a little to call `linear_out`?

Reply:
yup

Reply:
turns out you can't...

```
C++ exception with description "addmm(): functions with out=... arguments don't support automatic differentiation, but one of the arguments requires grad.
Exception raised from throw_error_out_requires_grad at ../torch/csrc/autograd/VariableTypeUtils.h:77 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x55 (0x7fba503adf55 in /data/users/mlevental/pytorch/build/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0xb1 (0x7fba503ab2c1 in /data/users/mlevental/pytorch/build/lib/libc10.so)
frame #2: <unknown function> + 0x2ec8dd9 (0x7fba5d32bdd9 in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #3: <unknown function> + 0x3090fd5 (0x7fba5d4f3fd5 in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #4: at::_ops::addmm_out::call(at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::Scalar const&, c10::Scalar const&, at::Tensor&) + 0x17f (0x7fba5bc2045f in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #5: at::native::linear_out(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, at::Tensor&) + 0xb6 (0x7fba5b672f56 in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #6: at::native::linear(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&) + 0x20f (0x7fba5b67332f in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #7: <unknown function> + 0x1bd5182 (0x7fba5c038182 in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #8: at::_ops::linear::call(at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&) + 0x13e (0x7fba5bcc0cbe in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #9: <unknown function> + 0x12cf550 (0x7fba5b732550 in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #10: <unknown function> + 0x12d24ec (0x7fba5b7354ec in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #11: at::native::gru_cell(at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::optional<at::Tensor> const&) + 0x141 (0x7fba5b7358f1 in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #12: <unknown function> + 0x1bd7d04 (0x7fba5c03ad04 in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #13: at::_ops::gru_cell::call(at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::optional<at::Tensor> const&, c10::optional<at::Tensor> const&) + 0x1af (0x7fba5bd39bef in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #14: torch::nn::GRUCellImpl::forward(at::Tensor const&, at::Tensor) + 0xcb (0x7fba5dfada0b in /data/users/mlevental/pytorch/build/lib/libtorch_cpu.so)
frame #15: ModulesTest_GRUCell_Test::TestBody() + 0x129 (0x8b6969 in build/bin/test_api)
frame #16: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) + 0x3a (0xbe24ea in build/bin/test_api)
frame #17: build/bin/test_api() [0xbd2bf8]
frame #18: build/bin/test_api() [0xbd301a]
frame #19: build/bin/test_api() [0xbd359d]
frame #20: testing::internal::UnitTestImpl::RunAllTests() + 0xbfa (0xbdb4ba in build/bin/test_api)
frame #21: testing::UnitTest::Run() + 0x7c (0xbdb93c in build/bin/test_api)
frame #22: main + 0xb8 (0x653c98 in build/bin/test_api)
frame #23: __libc_start_main + 0xf3 (0x7fba4fb63493 in /lib64/libc.so.6)
frame #24: _start + 0x2e (0x68df6e in build/bin/test_api)
" thrown in the test body.
[  FAILED  ] ModulesTest.GRUCell (8 ms)
```